sasl/sasl.go

```diff
@@ -42,3 +44,21 @@ type StateMachine interface {
 	// value will be true.
 	Next(ctx context.Context, challenge []byte) (done bool, response []byte, err error)
 }
+
+// Metadata contains additional data for performing SASL authentication.
+type Metadata struct {
+	Address string
```

Comment:
Could we document or rename this field? "Address" is a very generic term, it does not do a good job at explaining what value is expected here.

Reply:
I will both rename and document the field. I think a better name would be `BrokerAddress`